packages/contracts-bedrock/contracts/L2/L1Block.sol

```diff
@@ -74,5 +85,36 @@ contract L1Block is Semver {
         batcherHash = _batcherHash;
         l1FeeOverhead = _l1FeeOverhead;
         l1FeeScalar = _l1FeeScalar;
+
+        assembly {
+          sstore(add(_BLOCK_HASHES_STORAGE_OFFSET, mod(_number, BLOCK_HASHES_SIZE)), _hash)
+        }
+    }
+
+    /// @notice Returns the hash of the L1 block at the specified block number.
+    /// @dev Requires the block number to be within the range of stored block hashes.
+    /// @param _number The number of the block for which to return the hash.
+    /// @return The hash of the L1 block.
+    function getL1BlockHash(uint64 _number) external view returns (bytes32) {
+      uint64 _last_number = number;
+      bytes32 _hash;
+
+      if (_number > _last_number) {
+        revert L1BlockHashNumberOutOfBounds();
+      }
+
+      if (_last_number > BLOCK_HASHES_SIZE && _number <= _last_number - BLOCK_HASHES_SIZE) {
+        revert L1BlockHashNumberOutOfBounds();
+      }
+
+      assembly {
+        _hash := sload(add(_BLOCK_HASHES_STORAGE_OFFSET, mod(_number, BLOCK_HASHES_SIZE)))
+      }
+
+      if (_hash == bytes32(0)) {
```

Comment:
I think it is a good idea to add comment why _hash can ever be bytes32(0) given the two checks above